queries.sql

```diff
@@ -21,4 +21,75 @@ SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name!= 'Gabumon';
 
 
-SELECT * FROM animals WHERE weight_kg>= 10.4 AND weight_kg<=17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg>= 10.4 AND weight_kg<=17.3;
+
+
+/* Transactions */
+/* 1st Transaction */
+BEGIN;
+UPDATE animals SET species='unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+/* 2nd Transaction */
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT; 
+
+SELECT * FROM animals;
+
+
+/*3rd Transaction*/
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+
+/*4th Transaction*/
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SELECT * FROM animals;
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg=weight_kg*-1;
+SELECT * FROM animals;
+ROLLBACK sp1;
+SELECT * FROM animals;
+UPDATE animals SET weight_kg=weight_kg*-1 WHERE weight_kg<0;
+COMMIT;
+SELECT * FROM animals;
+
+
+/*queries for the question*/
+/*How many animals are there*/
+SELECT COUNT(*) FROM animals;
+
+/*How many animals have never tried to escape*/
+SELECT COUNT(*) FROM animals WHERE escape_attempts=0;
+
+
+/*What is the average weight of animals*/
+SELECT AVG(weight_kg) FROM animals;
+
+
+/*Who escapes the most, neutered or not neutered animals*/
+SELECT MAX(escape_attempts) FROM animals WHERE neutered=true;
+SELECT MAX(escape_attempts) FROM animals WHERE neutered=false;
+
+/*What is the minimum and maximum weight of each type of animal*/
+SELECT MAX(weight_kg), MIN(weight_kg) FROM animals;
+
+
+/*What is the average number of escape attempts per animal type of those born between 1990 and 2000*/
+SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth>'1990-01-01' AND date_of_birth<'2000-01-01';
+
```

Comment:
- [ ] Kindly update those three queries ( Who escapes the most, neutered or not neutered animals, What is the minimum and maximum weight of each type of animal, and What is the average number of escape attempts per animal type of those born between 1990 and 2000 ) to match the requirements and also please consider to do the `Who escapes the most, neutered or not neutered animals` in one query.
### Hint:
`Consider using group by statement`